### PR TITLE
Harvestable mechguns

### PR DESCRIPTION
**Pull request by anoobindisguise**
#### Summary
Content "Player-usable Mech Guns"

#### Purpose of change

I was really disappointed to find that there was no way to harvest the energy guns off of the mechsuits.
#### Describe the solution
I made it possible to harvest the energy guns off the mechsuits.
They are both vehicle-mounted heavy weapons as they were originally meant for the mechsuits. Can be installed on your vehicle turrets. Stats are the same as on their mechs.

#### Describe alternatives you've considered
Make the mechguns completely unable to fire unless they're mounted to a turret control unit since there might not be any "trigger" for them without their mech. There's no flag for this so I can't do it easily.

#### Testing
I strapped a gatling laser to my car and melted some goblins. Was fun.

### CODE REVIEW

**Comment by anoobindisguise** [comment]
something went fucky, reopening this in a sec.

**Comment by andrei8l** [comment]
You should be able to use `copy-from` for these new items; see the [documentation](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INHERITANCE.md).

**Comment by anoobindisguise** [comment]
> You should be able to use `copy-from` for these new items; see the [documentation](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INHERITANCE.md).

There are enough changes to them that I'm not sure if it's worth it but I'll take a look.

File: data/json/items/gun/ups.json [hunk 1]
```diff
@@ -190,5 +192,47 @@
...
     "flags": [ "NEVER_JAMS", "NO_UNLOAD", "NON_FOULING", "NEEDS_NO_LUBE" ]
+  },
+  {
+    "id": "gatling_mech_laser_single",
+    "copy-from": "gatling_mech_laser",
+    "looks_like": "m134",
+    "type": "GUN",
+    "name": { "str": "salvaged CMES laser cannon" },
+    "description": "This gatling-laser has been stripped from its CMES mech-suit and is far too heavy to use on its own.  You will need to deploy it or mount it to a vehicle, but it could be devastating if you have deep battery reserves.",
+    "valid_mod_locations": [
```
Comment (Maleclypse) [inline data/json/items/gun/ups.json:203]: Does this need valid mod locations?  Do mods work when attached to vehicles?

Comment (anoobindisguise) [reply data/json/items/gun/ups.json:203]: I *believe* they do. It's up in the air whether they should be moddable given their repurposed nature. Should I remove those entries?

Comment (Photoloss) [reply data/json/items/gun/ups.json:203]: Perhaps the availability could be checked more carefully. Fastening a carry strap to it should still be trivial and the laser components need to be serviced in some fashion even on the mech, but I doubt they provide standard sight rails or a typical stock attachment without those explicitly being crafted by the player for example. Your obtainment method currently implies just taking them from the mech as-is without any additional effort or resource expenditure to render them usable for a human operator.

**Comment by anoobindisguise** [comment]
Alright! Instead now you will need to use one of the "accessories" mod-addition mods such as the replaceable furniture mod to mod the mech guns, by default they accept accessories, laser mods (emitter and lens), and a sling mod.

**Comment by I-am-Erk** [comment]
I have a feeling these are not going to last as a feature (why use lasers instead of bullets on mechs) but I see no immediate barrier to this as long as the mechs are using energy guns.

**Comment by Terrorforge** [comment]
> (why use lasers instead of bullets on mechs)

I think there are lots of good reasons, actually. You've already got this honking great battery on it,  so it works out logistically. Probably saves weight. Reloading would be a pain. Hard to clear mechanical malfunctions from inside a mexj chassi. An ammo belt for a minigun would restrict the movement of the gun and risk getting tangled. It would also spit brass all over the place, possibly getting into the joints and such. And the recon model is supposed to be a stealth sniper, so obviously is makes sense to give it the quietest, most accurate weapon available.

**Pull request merged**